subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -120,14 +120,21 @@ static int rest_client_sckt_tls_setup(int fd, const char *const tls_hostname,
 	return 0;
 }
 
-static int rest_client_sckt_timeouts_set(int fd)
+static int rest_client_sckt_timeouts_set(int fd, int32_t timeout_ms)
 {
 	int err;
 	struct timeval timeout = { 0 };
 
 	if (CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT > 0) {
 		/* Send TO also affects TCP connect */
-		timeout.tv_sec = CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT;
+
+		/* TODO: CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT will never be used
+		 * because req_ctx->timeout_ms is set in rest_client_request_defaults_set()
+		 * Do we remove CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT and
+		 * CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT?
+		 */
+		timeout.tv_sec = timeout_ms > 0 ? timeout_ms / 1000 :
+			CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT;
```

Comment:
@jhirsi We need to discuss what to do here because `CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT` becomes irrelevant with this implementation when `rest_client_request_defaults_set` is used.

Reply:
Decided to remove `CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT` and `CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT` and use just `CONFIG_REST_CLIENT_REQUEST_TIMEOUT`.